src/Features/CSharp/Portable/EditAndContinue/StatementSyntaxComparer.cs

```diff
@@ -868,6 +886,11 @@ private static double ComputeWeightedDistance(ForStatementSyntax left, ForStatem
             {
                 distance = distance * 0.4 + localsDistance * 0.6;
             }
+            if (TryComputeLocalsDistance(left.Deconstruction, right.Deconstruction, out localsDistance))
```

Comment:
So I think you'll need to inline ```TryComputeLocalsDistance``` here and handle all possible combinations of nodes that declare locals.